modules/kubernetes-addons/csi-secrets-store-provider-aws/locals.tf

```diff
@@ -0,0 +1,30 @@
+locals {
+  name = "csi-secrets-store-provider-aws"
+
+  default_helm_config = {
+    name        = local.name
+    chart       = local.name
+    repository  = "https://aws.github.io/eks-charts"
+    version     = "0.0.2"
+    namespace   = local.name
+    description = "A Helm chart to install the Secrets Store CSI Driver and the AWS Key Management Service Provider inside a Kubernetes cluster."
+    values      = []
+    timeout     = "1200"
+  }
+
+  helm_config = merge(
+    local.default_helm_config,
+    var.helm_config
+  )
+
+  irsa_config = {
+    create_kubernetes_namespace       = true
+    kubernetes_namespace              = local.name
+    create_kubernetes_service_account = false
+    kubernetes_service_account        = local.name
+  }
```

Comment:
Don't use irsa module if we are not creating an IAM role with service account. 
Simply create a Namespace object for this

Reply:
@vara-bonthu - I was creating a namespace object earlier, but i was asked to use this instead per https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/471#discussion_r863312265 . Tagging @askulkarni2 for visibility. Pls lmk a final direction that you both align on.

Reply:
@anshrma Please go-head and change this to simple namespace resource. Use IRSA module only when you need IAM role for service accounts